### ISSUE TITLE
make parameters more reasonable

### ISSUE THREAD

**Issue by nolouch** [tikv/pd#3749]
### Describe your feature request related problem

We have encountered many problems caused by improper parameter adjustment. We must limit the adjustment range of the parameters and give a best practice, remove some useless and do some check.

Such as: `merge-limit=128, region-limit=64, leader-limit=8` the merge limit may influence the leader limit.

### Describe alternatives you've considered

- make the limit do not affect each other
- limit the adjustment range, such as `store limit` should have best practice
- do some check if the user set parameters is unreasonble

### Teachability, Documentation, Adoption, Migration Strategy

- GO

**Comment by xuhui-lu** [comment]
hey @nolouch, may I pick up this issue?

**Comment by xuhui-lu** [comment]
And I think the limit here means the parameters for scheduler, is my understanding correct?

**Comment by nolouch** [comment]
@xuhui-lu  Yes, you are right. such as `store limit`. we need to set a reasonable range for some parameters. Do you have any ideas?
PTAL @rleungx @bufferflies

**Comment by bufferflies** [comment]
I think you should decouple the merge-limit, region-limit, leader-limit first.

**Comment by bufferflies** [comment]
The relation between limit and scheduler(checker) is below:
![image](https://user-images.githubusercontent.com/23159587/123610814-8880b480-d833-11eb-90b2-f0b0ab07b108.png)

the relation between opKind and scheduler(checker) is below:
![image](https://user-images.githubusercontent.com/23159587/123611171-e90ff180-d833-11eb-9fc4-33bb5cfcb394.png)

**Comment by bufferflies** [comment]
There are some job to solve this problem: 

- [ ] one operator only consume one limit, it will consume leader limit if operator include  transfer leader and move peer
- [ ] add leader store limit to modify one store's leader limit independent.
